### PR TITLE
fix!: add status as a number to GaxiosError, change code to error code as a string

### LINKED ISSUE #1092: err.code type changes from 'string' to 'number'.
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

1) Is this a client library issue or a product issue?
This is the client library for . We will only be able to assist with issues that pertain to the behaviors of this library. If the issue you're experiencing is due to the behavior of the product itself, please visit the [ Support page]() to reach the most relevant engineers.

2) Did someone already solve this?
  - Search the issues already opened: https://github.com/googleapis/google-auth-library-nodejs/issues
  - Search the issues on our "catch-all" repository: https://github.com/googleapis/google-cloud-node
  - Search or ask on StackOverflow (engineers monitor these tags): http://stackoverflow.com/questions/tagged/google-cloud-platform+node.js

3) Do you have a support contract?
Please create an issue in the [support console](https://cloud.google.com/support/) to ensure a timely response.

If the support paths suggested above still do not result in a resolution, please provide the following details.

#### Environment details

  - OS: Windows 10.0.18362
  - Node.js version: 12.13.0
  - npm version: 6.12.0
  - `google-auth-library` version: 6.1.1.

#### Steps to reproduce

  1. Call google.classroom("v1").courses.list(), or any related API (courses.teacher, courses.student, etc.)
  2. Keep calling until a quota threshold is reached (should get a 429 error code).
  3. Look at the typeof the error.code returned.  It will be a 'number', but should be a 'string'.

Looks like 
   err.code = body.error.code; 
around line 95 in processError() in transporters.js needs to ensure the assigned value is a string.

Line 112 handles this by calling toString().  Might be other lines that need to ensure the code field is a string type.

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

### PR DESCRIPTION
**Pull request by sofisl**
Fixes: https://github.com/googleapis/google-auth-library-nodejs/issues/1092

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -17,19 +17,24 @@ import {URL} from 'url';
...
   ) {
     super(message);
     this.response = response;
     this.config = options;
     this.response.data = translateData(options.responseType, response.data);
-    this.code = response.status.toString();
+    if (error && 'code' in error && error.code) {
+      this.code = Number(error.code);
+    }
+    this.status = response.status.toString();
```
Comment (danielbankhead) [inline src/common.ts:37]: I think this should be the `status`, without converting to the string to match this convention:
https://developer.mozilla.org/en-US/docs/Web/API/Response/status

File: src/common.ts [hunk 2]
```diff
@@ -17,19 +17,24 @@ import {URL} from 'url';
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
 export class GaxiosError<T = any> extends Error {
-  code?: string;
+  code?: Number;
```
Comment (danielbankhead) [inline src/common.ts:20]: I think this should be a `string` to match this convention:
https://nodejs.org/api/errors.html#errorcode

File: src/common.ts (hunk 1 above)
Comment (sofisl) [reply src/common.ts]: oops, clearly I just mixed them up!

File: src/common.ts [hunk 3]
```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
...
+    response: GaxiosResponse<T>,
+    error?: Error | NodeJS.ErrnoException
   ) {
     super(message);
     this.response = response;
     this.config = options;
     this.response.data = translateData(options.responseType, response.data);
-    this.code = response.status.toString();
+    if (error && 'code' in error && error.code) {
+      this.code = error.code.toString();
```
Comment (danielbankhead) [inline src/common.ts:35]:
```suggestion
      this.code = error.code;
```

File: src/common.ts [hunk 4]
```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
...
+    }
+    this.status = Number(response.status);
```
Comment (danielbankhead) [inline src/common.ts:37]:
```suggestion
    this.status = response.status;
```

File: src/common.ts [hunk 5]
```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
   code?: string;
```
Comment (danielbankhead) [inline src/common.ts:20]: Suggesting some doc for usage/clarity.

```suggestion
  /**
  * An Error code.
  * See {@link https://nodejs.org/api/errors.html#errorcode error.code}
  *
  * @example
  * 'ECONNRESET'
  */
  code?: string;
```

File: src/common.ts [hunk 6]
```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
...
   response?: GaxiosResponse<T>;
   config: GaxiosOptions;
+  status: Number;
```
Comment (danielbankhead) [inline src/common.ts:23]: Doc + type change. There are a few differences between [`number` and `Number`](https://stackoverflow.com/questions/67155108/what-is-the-difference-between-number-and-number-in-typescript)

```suggestion
  /**
  * An HTTP Status code.
  * See {@link https://developer.mozilla.org/en-US/docs/Web/API/Response/status Response: status property}
  *
  * @example
  * 500
  */
  status: number;
```

**Review by danielbankhead — COMMENTED** [review]
Looks good, a few suggestions

File: src/common.ts [hunk 7]
```diff
@@ -20,16 +20,21 @@ export class GaxiosError<T = any> extends Error {
...
   constructor(
     message: string,
     options: GaxiosOptions,
-    response: GaxiosResponse<T>
+    response: GaxiosResponse<T>,
+    error?: Error | NodeJS.ErrnoException
```
Comment (danielbankhead) [inline src/common.ts:28]: Lastly, we can expose the provided error if someone wants to debug further
```suggestion
    public error?: Error | NodeJS.ErrnoException
```

**Review by danielbankhead — APPROVED** [review]

**Pull request merged**